packages/flutter/test/gestures/scale_test.dart

```diff
@@ -1356,4 +1356,296 @@ void main() {
 
     scale.dispose();
   });
+
+  testGesture('ScaleStartDetails and ScaleUpdateDetails callbacks should contain thier event.timestamp', (GestureTester tester) {
+    final ScaleGestureRecognizer scale = ScaleGestureRecognizer();
+    final TapGestureRecognizer tap = TapGestureRecognizer();
+
+    bool didStartScale = false;
+    Offset? updatedFocalPoint;
+    Duration? initialSourceTimestamp;
+    scale.onStart = (ScaleStartDetails details) {
+      didStartScale = true;
+      updatedFocalPoint = details.focalPoint;
+      initialSourceTimestamp = details.sourceTimeStamp;
+    };
+
+    double? updatedScale;
+    double? updatedHorizontalScale;
+    double? updatedVerticalScale;
+    Offset? updatedDelta;
+    Duration? updatedSourceTimestamp;
+    scale.onUpdate = (ScaleUpdateDetails details) {
+      updatedScale = details.scale;
+      updatedHorizontalScale = details.horizontalScale;
+      updatedVerticalScale = details.verticalScale;
+      updatedFocalPoint = details.focalPoint;
+      updatedDelta = details.focalPointDelta;
+      updatedSourceTimestamp = details.sourceTimeStamp;
+    };
+
+    bool didEndScale = false;
+    scale.onEnd = (ScaleEndDetails details) {
+      didEndScale = true;
+    };
+
+    bool didTap = false;
+    tap.onTap = () {
+      didTap = true;
+    };
+
+    final TestPointer pointer1 = TestPointer();
+
+    final PointerDownEvent down = pointer1.down(Offset.zero, timeStamp: const Duration(milliseconds: 10));
+    scale.addPointer(down);
+    tap.addPointer(down);
+
+    tester.closeArena(1);
+    expect(didStartScale, isFalse);
+    expect(updatedScale, isNull);
+    expect(updatedFocalPoint, isNull);
+    expect(updatedDelta, isNull);
+    expect(updatedSourceTimestamp, isNull);
+    expect(didEndScale, isFalse);
+    expect(didTap, isFalse);
+    expect(initialSourceTimestamp, isNull);
+
+    // One-finger panning
```

Comment:
consider ending these comments with a period, here and in other places.

Reply:
These are copied from the origional tests. Should those be changed as well?

Reply:
Not necessary for this change, but would be welcome.

Reply:
done.